### DISCUSSION TITLE
How to bulk upload files to storage

### DISCUSSION THREAD

**Discussion by cspecter** [supabase/supabase#1691] [Questions] [answered]
I have around 16,000 images that I want to put on storage. Is there a convenient way to do this? We have tried through the Supabase web dashboard and have not been able to bulk upload that many files.

Update: We did get most to upload from the web interface. It just locked up the browser for a while with no feedback.

Side question: Is there a way to delete a folder and all it's contents?

**Comment by roryw10** [comment, marked as answer]
You can use the API directly to do this and run a script. 

https://supabase.github.io/storage-api/#/

**Reply by cspecter** [reply]
FYI, we got the web interface to work. It locked up for a while, but ultimately chugged through all the files.
